lib/iris/experimental/ugrid.py

```diff
@@ -615,17 +615,130 @@ def equal(self, other, lenient=None):
         return super().equal(other, lenient=lenient)
 
 
-#: Convenience collection of lenient metadata combine services.
-SERVICES_COMBINE.append(ConnectivityMetadata.combine)
-SERVICES.append(ConnectivityMetadata.combine)
+class MeshCoordMetadata(BaseMetadata):
+    """
+    Metadata container for a :class:`~iris.coords.MeshCoord`.
 
+    """
 
-#: Convenience collection of lenient metadata difference services.
-SERVICES_DIFFERENCE.append(ConnectivityMetadata.difference)
-SERVICES.append(ConnectivityMetadata.difference)
+    _members = ("mesh", "location", "axis")
```

Comment:
@pp-mo Whoa... are you really suggesting that `mesh` is metadata here?

This is the payload of the `MeshCoord`, right? If so, then I don't agree with this... it's akin to making `points` and `bounds` members of the `CoordMetadata` or `DimCoordMetadata` :-1: 

Or do you mean `mesh` to be something different?

Reply:
Obviously having the actual `Mesh` as part of the metadata wouldn't work, but given the idea of shared `Mesh`es I'd say some sort of mesh identifier is an important component of the `MeshCoord` metadata.

Reply:
If `mesh` is a cheap comparison, then yes this could be part of the metadata, but until we know better I think we should err on the side of this not being part the metadata

Reply:
Also, `__eq__` hasn't been defined within `iris.experimental.ugrid.Mesh` for this very reason... we're kinda sitting on the fence until we know better

Reply:
Ok removed it